R/git_tools.R

```diff
@@ -174,10 +175,18 @@ install_external_deps <- function(repo_dir, internal_pkg_deps, ...) {
   # remove internal_pkg_deps from DESCRIPTION file
   desc_obj <- desc::desc(file.path(repo_dir_external, "DESCRIPTION"))
   new_deps <- desc_obj$get_deps()[!desc_obj$get_deps()$package %in% internal_pkg_deps,]
+
+  #also change name to make it clearer for users when using renv::install
+  desc_obj$set(Package = "staged.deps.tmp.pkg")
   desc_obj$set_deps(new_deps)
   desc_obj$write()
 
-  remotes::install_deps(repo_dir_external, ...)
+  if (!is.null(Sys.getenv("RENV_PROJECT")) && Sys.getenv("RENV_PROJECT") != "" && requireNamespace("renv", quietly = TRUE)) {
+    renv::install(repo_dir_external)
+    renv::remove("staged.deps.tmp.pkg")
```

Comment:
May be able to remove line 180 and 186 and it still work and be less confusing for users?

Reply:
yup these can and have been removed